File: app/src/main/java/co/medellinandroid/spotifyartists/rest/ApiInterceptor.java
package co.medellinandroid.spotifyartists.rest;

import java.io.IOException;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

import static co.medellinandroid.spotifyartists.utils.Constants.ACCESS_TOKEN;
import static co.medellinandroid.spotifyartists.utils.Constants.API_KEY;

public class ApiInterceptor implements Interceptor {

  @Override public Response intercept(Chain chain) throws IOException {

    Request original = chain.request();
    Request request = original.newBuilder()
        .header(API_KEY, ACCESS_TOKEN)
        .method(original.method(), original.body())
        .build();
    return chain.proceed(request);
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/rest/ApiInterceptor.kt
package co.medellinandroid.spotifyartists.rest

import co.medellinandroid.spotifyartists.utils.Constants.Companion.ACCESS_TOKEN
import co.medellinandroid.spotifyartists.utils.Constants.Companion.API_KEY
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ApiInterceptor : Interceptor {

  @Throws(IOException::class)
  override fun intercept(chain: Interceptor.Chain): Response {

    val original = chain.request()
    val request = original.newBuilder()
        .header(API_KEY, ACCESS_TOKEN)
        .method(original.method(), original.body())
        .build()
    return chain.proceed(request)
  }
}
